src/index.ts

```diff
@@ -440,28 +440,13 @@ export function getBootloaderSync() {
   return bootloader;
 }
 
-let device: string;
-export async function getDevice() {
-  if (!device) {
-    if (Platform.OS === 'android') {
-      device = await RNDeviceInfo.getDevice();
-    } else {
-      device = 'unknown';
-    }
-  }
-  return device;
-}
-
-export function getDeviceSync() {
-  if (!device) {
-    if (Platform.OS === 'android') {
-      device = RNDeviceInfo.getDeviceSync();
-    } else {
-      device = 'unknown';
-    }
-  }
-  return device;
-}
+export const [getDevice, getDeviceSync] = getSupportedPlatformInfoFunctions({
+  getter: () => RNDeviceInfo.getDeviceName(),
+  syncGetter: () => RNDeviceInfo.getDeviceNameSync(),
```

Comment:
I think you intend these native getters for this one?

```suggestion
  getter: () => RNDeviceInfo.getDevice(),
  syncGetter: () => RNDeviceInfo.getDeviceSync(),
```

Reply:
good catch!